guides/docs/channels.md

```diff
@@ -1,60 +1,127 @@
 # Channels
 
-Channels are a really exciting and powerful part of Phoenix that allow us to easily add soft-realtime features to our applications. Channels are based on a simple idea - sending and receiving messages. Senders broadcast messages about topics. Receivers subscribe to topics so that they can get those messages. Senders and receivers can switch roles on the same topic at any time.
-
-Since Elixir is based on message passing, you may wonder why we need this extra mechanism to send and receive messages. With Channels, neither senders nor receivers have to be Elixir processes. They can be anything that we can teach to communicate over a Channel - a JavaScript client, an iOS app, another Phoenix application, our watch. Also, messages broadcast over a Channel may have many receivers. Elixir processes communicate one to one.
-
-The word "Channel" is really shorthand for a layered system with a number of components. Let's take a quick look at them now so we can see the big picture a little better.
-
-## JS Documentation
-
-Phoenix ships with a JavaScript client that is available when generating a new Phoenix project. The documentation for the JavaScript module is available at [https://hexdocs.pm/phoenix/js/](https://hexdocs.pm/phoenix/js/)
+Channels are an exciting part of Phoenix that enable soft real-time communication with and between millions of connected clients.
+Some possible use cases include:
+
+- Chat rooms
+- Breaking news, like "a goal was scored" or "an earthquake is coming"
+- Events in multiplayer games
+- Monitoring sensors and controlling lights
+- Notifying a browser that a page's CSS or JavaScript has changed (this is handy in development)
+
+Conceptually, Channels are pretty simple.
+Clients connect and subscribe to one or more topics, whether that's `public_chat` or `updates:user1`.
+Any message sent on a topic, whether from the server or from a client, is sent to all clients subscribed to that topic, like this:
+
+<pre>
+<code>
```

Comment:
I put in these `<pre><code>` tags manually because if I just indented, the resulting `<code>` tags had `highlightjs` and tried to highlight the diagrams as Elixir, with inconsistent coloring.